commons/java/openjdk11/Dockerfile.arm64

```diff
@@ -0,0 +1,6 @@
+FROM cloudsuitetest/debian:base-os
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends openjdk-11-jdk-headless \
+        && rm -rf /var/lib/apt/lists/*
+# TODO: Fix this link for other architectures.
+ENV JAVA_HOME /usr/lib/jvm/java-11-openjdk-arm64/
```

Comment:
I'm pretty sure you can do this in a better way by getting the architecture from the system and then set this variable based on that instead of having three multiple files.

There might be another way of doing this specifically with Java because I think you can find the Java home path using a command, but the first approach should work for sure.

Reply:
@Xusine1131 Can you check for a better way to set JAVA_HOME?

Reply:
@neo-apz The current version of diff-set has failed because the source and target branches are on different forks. If they reside on the same branch, I believe it will work fine. I try to see whether I can implement it more efficiently.

Reply:
Yeah but we want that to work as well. I haven't looked at your code much, but I just saw that you are trying to log in before checking the diff-set, that by itself to me sounds like an issue, because this should not rely on logging in at all. And most likely this is what's causing the issue.

In any case, as mentioned before, if using a pre-created GHA means a lot of duplicated code, then we don't want to do thing like that and you should create your own scripts (like what we had before).

Reply:
Would you give me the pointer to where I login before checking the diff-set?

Reply:
You have this line, which I assume is used for authorization:

```
GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Also if you look at the error message it's clear it assumes the other branch is a part of this repo and not another fork, here is the link it's trying to use for comparison:
```
https://api.github.com/repos/parsa-epfl/cloudsuite/compare/dev...fix_issue_290
```

Reply:
> I'm pretty sure you can do this in a better way by getting the architecture from the system and then set this variable based on that instead of having three multiple files.
> 
> There might be another way of doing this specifically with Java because I think you can find the Java home path using a command, but the first approach should work for sure.

@neo-apz I don't think current Docker support set ENV with a value from a command line. Please check this issue: https://github.com/moby/moby/issues/29110

Reply:
Two things:
1. why can't you just write a simple if-else which depending on the result of a command that gives you the architecture (e.g., `arch` or `uname`) sets the correct path? 

2. There are other ways of setting the environment variable, check this for example: https://stackoverflow.com/questions/34911622/dockerfile-set-env-to-result-of-command